### PR TITLE
Improve monthly graph on small screens

### PR DESCRIPTION
**Pull request by michaelrambeau**
## Goal

Make the monthly graphs displayed in the project details page better on small screens.

Currently the numbers at the top and the months at the bottom are not displayed correctly.

image

## Changes

- The number at the top of the bar are no more displayed on small screens
- A tooltip is displayed when the users "hovers" the bar
- The  month and the year are displayed in a different way (see screenshot)

## How to test

Open the project detail page `/projects/:id` on a small screen and check that both graphs are displayed correctly:

- Stars added on GitHub, month by month
- NPM monthly downloads
 
A unit has been added to check the logic about the component that displays the year number at the bottom of the graph.

### On small screens

image

image

### On bigger screens

image

image

### CODE REVIEW

File: src/components/project-details/monthly-trends-graph/monthly-trends-chart.tsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  const formattedValue = formatValue(value, { decimals: 1 });
+  const formattedDate = year + "/" + month;
+  const tooltipLabel = formattedDate + ": " + formattedValue + " " + unit;
+
+  return (
+    <Box
+      position="relative"
+      h={height}
+      className="hint--top"
+      aria-label={tooltipLabel}
```
Comment (michaelrambeau) [inline src/components/project-details/monthly-trends-graph/monthly-trends-chart.tsx:106]: Note: the tooltip works with Hint.css https://kushagra.dev/lab/hint/

**Review by nvh95 — APPROVED** [review]
LGTM!
I think it's nice to have if you can add some integration tests to verify the content of the project detail. We haven't had it yet. The scenario is:
- User navigates to a project
- They should see some correct numbers showed up (Please get number from `npm run start:mock`)
By doing this. In the future, if you update statistic portions, you will have confidence that new code works fine.

This is completely optional. Feel free to skip it.

**Comment by michaelrambeau** [comment]
## Next iteration

image

image

image

image

**Comment by michaelrambeau** [comment]
## Behavior on mobiles

On mobiles, clicking on the bar shows the numbers, since there is no enough space to show all the numbers.

image

**Comment by michaelrambeau** [comment]
## Final version

The latest version restores the `+` symbol used to show the number of stars added by months, following a comment from Sacha on Slack:

> I like it but I miss the "+" to make it really clear that it's stars added and not the total number of stars

image

**Comment by michaelrambeau** [comment]
> LGTM! I think it's nice to have if you can add some integration tests to verify the content of the project detail. We haven't had it yet. The scenario is:
> 
>     * User navigates to a project
> 
>     * They should see some correct numbers showed up (Please get number from `npm run start:mock`)
>       By doing this. In the future, if you update statistic portions, you will have confidence that new code works fine.
> 
> 
> This is completely optional. Feel free to skip it.

Thank you for your comments Hung!
It will be handled in another PR as I think we need some changes in the mocking strategy.
The monthly graphs expect to get data from the 12 previous months but when using the mock mode we get data related to a specific date, when tests were created, in April, which leads to "holes" in the graph for the months after April.

**Pull request merged**